src/test/java/org/example/controller/EmployeeControllerTest.java

```diff
@@ -0,0 +1,145 @@
+                .andExpect(jsonPath("$.gender").value("Male"))
+                .andExpect(jsonPath("$.dateOfBirth").value("1990-01-01"))
+                .andExpect(jsonPath("$.graduationDate").value("2012-06-15"))
+                .andExpect(jsonPath("$.departmentId").value(1))
+                .andExpect(jsonPath("$.managerId").value(nullValue()))  // Ensure managerId is null
+                .andExpect(jsonPath("$.teamId").value(1))
+                .andExpect(jsonPath("$.expertise[0]").value("Java"))  // Check the expertise array
+                .andExpect(jsonPath("$.expertise[1]").value("Spring Boot"))
+                .andExpect(jsonPath("$.grossSalary").value(75000.0));
+    }
+
+    @Test
+    public void testCreateEmployeeWithInvalidData() throws Exception {
+        // Missing name field
+        mockMvc.perform(post("/employees")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(String.format("""
+                        {
+                            "gender": "Male",
+                            "dateOfBirth": "1990-01-01",
+                            "graduationDate": "2012-06-15",
+                            "expertise": "Software Development",
+                            "grossSalary": 60000.0,
+                            "departmentId": %d
+                        }
+                        """, testDepartmentId)))
+                .andExpect(status().isBadRequest());
+
+        // Invalid graduation date
+        mockMvc.perform(post("/employees")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(String.format("""
+                        {
+                            "name": "Jane Doe",
+                            "gender": "Female",
+                            "dateOfBirth": "1990-01-01",
+                            "graduationDate": "invalid-date",
+                            "expertise": "Software Development",
+                            "grossSalary": 50000.0,
+                            "departmentId": %d
+                        }
+                        """, testDepartmentId)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    public void testGetEmployeeById() throws Exception {
+        // Create employee first
+        String employeeJson = String.format("""
+                        {
+                            "name": "John Doe",
+                            "gender": "Male",
+                            "dateOfBirth": "1990-01-01",
+                            "graduationDate": "2012-06-15",
+                            "expertise": "Software Development",
+                            "grossSalary": 60000.0,
+                            "departmentId": %d
+                        }
+                        """, testDepartmentId);
+
+        // Create employee and extract the response
+        String response = mockMvc.perform(post("/employees")
```

Comment:
no no no , y3ni hwa keda keda hndt5dm database setup bs 7ta lw ht3mlha keda w aly hwa l2 bs we never callanother api in a test for an api
y3ni anta dlw2ty bt3ml test ll get api mynf3sh t call l save api 5ales
3shan l test bta3ak lazem ykon independent 
mynf3sh y fail bss an l save 3mlt fail

Reply:
plus handle all negative scnarios for all apis please